### PR TITLE
[K32W1] Initial enablement

### PR DESCRIPTION
**Pull request by marius-alex-tache**
Add support for K32W1 MCU: https://www.nxp.com/products/wireless/multiprotocol-mcus/secure-and-ultra-low-power-mcu-for-matter-over-thread-and-bluetooth-le-5-3:K32W148

### CODE REVIEW

**Review by George-Stefan — APPROVED** [review]

**Review by andrei-menzopol — APPROVED** [review]

**Review by doru91 — APPROVED** [review]

**Comment by marius-alex-tache** [comment]
Do not merge! Some debugging is currently going on. We need to update the `cmake` infrastructure as well, to avoid having two different version on the public repos.

**Comment by abtink** [comment]
> Do not merge! Some debugging is currently going on. We need to update the `cmake` infrastructure as well, to avoid having two different version on the public repos.

Added "work-in-progress" label to PR to help indicate this, we can remove it later once ready.

**Comment by marius-alex-tache** [comment]
From my point of view, we can merge this.

File: cmake/utils.cmake [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2022, NXP.
```
Comment (jwhui) [inline cmake/utils.cmake:1]: Change to OpenThread header?

**Comment by George-Stefan** [comment]
> From my point of view, we can merge this.

@andrei-menzopol, can you handle it?

File: cmake/utils.cmake (hunk 1 above)
Comment (andrei-menzopol) [reply cmake/utils.cmake:1]: Sure

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**